vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/TemporalDataReader.java

```diff
@@ -0,0 +1,275 @@
+        List<VortexGrid> relevantGrids = recordIndexQuery.query(startTime, endTime).stream()
+                .map(bufferedReader::get)
+                .sorted(Comparator.comparing(VortexGrid::startTime))
+                .toList();
+
+        if (startTime.isEqual(endTime)) {
+            float[] data = TemporalDataCalculator.calculatePointInstant(relevantGrids, startTime);
+            return TemporalDataCalculator.buildGrid(bufferedReader.getBaseGrid(), startTime, startTime, data);
+        } else {
+            float[] data = TemporalDataCalculator.calculatePeriodInstant(relevantGrids, startTime, endTime);
+            return TemporalDataCalculator.buildGrid(bufferedReader.getBaseGrid(), startTime, endTime, data);
+        }
+    }
+
+    /* Get Min & Max Methods */
+    private VortexGrid[] getMinMaxGridData(VortexDataType dataType, ZonedDateTime startTime, ZonedDateTime endTime) {
+        return switch (dataType) {
+            case ACCUMULATION, AVERAGE -> getMinMaxForPeriodGrids(startTime, endTime);
+            case INSTANTANEOUS -> getMinMaxForInstantGridsWithRecordQuery(startTime, endTime);
+            default -> null;
+        };
+    }
+
+    private VortexGrid[] getMinMaxForPeriodGrids(ZonedDateTime startTime, ZonedDateTime endTime) {
+        List<Integer> indices = new ArrayList<>(recordIndexQuery.query(startTime, endTime));
+        List<VortexGrid> relevantGrids = indices.stream().map(bufferedReader::get).toList();
+        return buildMinMaxGrids(relevantGrids, startTime, endTime);
+    }
+
+    private VortexGrid[] getMinMaxForInstantGridsWithRecordQuery(ZonedDateTime startTime, ZonedDateTime endTime) {
+        List<Integer> indices = new ArrayList<>(recordIndexQuery.query(startTime, endTime));
+        indices.remove(indices.size() - 1); // Drop last index
+
+        List<VortexGrid> relevantGrids = indices.stream()
+                .map(bufferedReader::get)
+                .filter(g -> g.startTime().toEpochSecond() >= startTime.toEpochSecond())
+                .toList();
+        return buildMinMaxGrids(relevantGrids, startTime, endTime);
+    }
+
+    private VortexGrid[] buildMinMaxGrids(List<VortexGrid> grids, ZonedDateTime startTime, ZonedDateTime endTime) {
+        float[][] minMaxData;
+
+        if (grids.isEmpty()) {
+            minMaxData = new float[][] {new float[0], new float[0]};
+        } else {
+            minMaxData = TemporalDataCalculator.getMinMaxForGrids(grids);
+        }
+
+        float[] minData = minMaxData[0];
+        float[] maxData = minMaxData[1];
+
+        VortexGrid minGrid = TemporalDataCalculator.buildGrid(bufferedReader.getBaseGrid(), startTime, endTime, minData);
+        VortexGrid maxGrid = TemporalDataCalculator.buildGrid(bufferedReader.getBaseGrid(), startTime, endTime, maxData);
+
+        return new VortexGrid[] {minGrid, maxGrid};
+    }
+
+    private List<VortexGrid> getRelevantGrids(ZonedDateTime startTime, ZonedDateTime endTime) {
```

Comment:
Maybe rename to something more explicit like `getGridsForPeriod(...`

Reply:
Renamed